### ISSUE TITLE
handle suspicious votes

### ISSUE THREAD

**Issue by romaindurand** [romaindurand/maju.app#15]
- [ ] warn users when there are suspicious votes
  (when an ip voted more than once)
- [ ] display stats about suspicious votes to user
- [ ] allow user to see 'cleaned' results

**Comment by oceanBigOne** [comment]
Salut, 

je me suis un peu posé la question sur ce sujet. De mon expérience, il arrive que plusieurs personnes vote depuis la même ip sans qu’il y ai de « tricherie ». Ce qui m’est arrivé c’est un vote lors d’une soirée sur téléphone portable avec beaucoup de téléphone sur le même wifi. J’ai aussi eu le cas, d’un vote ou les utilisateurs se font passer le même téléphone pour voter chacun leur tour :/ ...

Du coup, je ferai deux remarques :
1/ peut-être associer un cookie et/ou la détection des entêtes de navigateur en plus de l’ip
2/ avoir un message pas trop anxiogène et expliquer que cela peut-être normal
3/ laisser l’utilisateur  saisir un pseudo/nom au moment du vote

Pour le point 3, ca permet, sur un petit vote entre personnes qui se connaissent de valider la liste des participants.

**Comment by romaindurand** [comment]
Oui, l'identification des votants, j'y songe aussi, mais j'essaie de repousser ça le plus possible pour l'expérience utilisateur : moins il a à faire de choses, plus il a de chances de s'en servir fréquemment.

Mon modèle (totalement assumé, tu peux voir les ressemblances xD) c'est [strawpoll](https://strawpoll.com/) (vote par approbation), qui est utilisé entre autre par beaucoup de streameurs de jeux vidéo qui demandent par exemple à leur spectateurs à quel jeu il va jouer après : de l'interaction rapide avec sa communauté en gros. Ils touchent un public large et encore peu conditionné par le scrutin uninominal, à qui j'aimerais faire découvrir le jugement majoritaire.

Je compte pousser l'idée via des communautés actives, qui recoupent cette première, et qui pourraient en faire la promotion, comme le discord de youtubeurs politique/humour/actu ou le discord insoumis, souvent assez friands de ce genre d'initiatives.

Mais sans identifier l'utilisateur (pseudonyme, voire vérification par email), j'essaie de trouver un compromis entre la sécurité/fiabilité du résultat et la facilité d'utilisation :

Sur les sécurités en place, j'utilise déjà :
- les cookies
- le local-storage
- une protection par Recaptcha pour éviter des votes automatisés

Et pour chaque vote, j'enregistre l'IP et une [fingerprint](http://valve.github.io/fingerprintjs2/)

Rien qui ne soit contournable pour quelqu'un de déterminé, mais ça reste suffisant pour beaucoup de cas d'usage, je compte plus sur le projet de l'asso ou le tien pour proposer un système plus sécurisé, qui serait utilisé pour des scrutins plus officiels, des usages complémentaires quoi 😉 

**Je note bien ton point sur l'importance du coté non-anxiogène du message d'alerte, c'est vrai que ça peut être légitime !**

Pour la possibilité de laisser un pseudo, c'est sans doute une option que j'ajouterai, mais plus tard. Je n'y ai pas encore beaucoup réfléchi, mais j'ai l'impression qu'il faut cacher le nom des participants avant la clôture des votes (donc date de fin requise), sans quoi on risque de perdre l'anonymat
